# Graded Challenge 3

***
Nama  : Catherine Kezia Wijaya <br>
Batch : RMT-037
***

## ii. Identifikasi Masalah

### A. Case

Seblak adalah makanan yang populer di kalangan masyarakat. Banyak orang membicarakannya terutama di media sosial, sehingga muncul asumsi bahwa minat terhadap produk seblak tinggi. Di tengah trend ini, skema dropship di platform e-commerce seperti Tokopedia menjadi salah satu bentuk bisnis tanpa perlu modal yang besar untuk produksi barang. 

Namun, sebelum memulai bisnis dropship seblak, penting untuk melakukan analisis data terhadap trend penjualan seblak di Tokopedia untuk memahami apakah produk ini benar-benar memiliki banyak minat di masyarakat seperti jumlah penjualan, rating dan harga produk.

Sebagai lulusan Bootcamp Data Science Hacktiv8, kemampuan dalam analisis data sangat berguna dalam mengambil jalan skema bisnis ini. Meskipun tidak memiliki akses langsung ke data penjualan Tokopedia, kemampuan web scraping dapat digunakan untuk mengumpulkan data yang tersedia di platform. Dengan begitu, data yang didapat dari web scrapping bisa dianalisis untuk memahami tren dan potensi pasar seblak di Tokopedia.

### B. Objective

- Mengumpulkan data penjualan seblak dari Tokopedia melalui web scraping, termasuk informasi nama produk, harga produk, jumlah penjualan, asal kota penjual dan rating..
- Menganalisis faktor - faktor yang dapat mempengaruhi minat customer terhadap produk.
- Memberikan insight bisnis berdasarkan analisis data untuk memutuskan apakah seblak merupakan produk yang tepat untuk dijual dengan skema dropship di Tokopedia.

## 2. Answer

### A. Web Scrapping

- Diambil halaman yang ditampilkan Tokopedia setelah memasukan keyword seblak ke dalam search engine
- Data yang diambil adalah nama produk, harga produk, nama penjual, kota asal penjual, dan rating produk
- Diambil 10 halaman pertama

In [2]:
from selenium import webdriver
from bs4 import BeautifulSoup
import time

In [3]:
nama_produk = []
harga_produk = []
list_penjual = []
kota_toko = []
banyak_terjual = []
rating_produk = []

driver = webdriver.Chrome() 

# Ambil 10 halaman data
for i in range(1,11):
# Inisialisasi WebDriver
    
    url = 'https://www.tokopedia.com/search?navsource=&page={}&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st='.format(i)
    driver.get(url)

    for i in range(19):
        driver.execute_script("window.scrollBy(0, 250)")
        time.sleep(1)

    driver.execute_script("window.scrollBy(50,0)")
    time.sleep(5)

    # Ambil konten halaman setelah dirender oleh JavaScript
    html = driver.page_source

    # Parse dengan BeautifulSoup
    soup = BeautifulSoup(html, 'html.parser')

    # Menyimpan semua produk yang memiliki class `css-5wh65g` dalam satu list `list_produk`
    list_produk = soup.find_all('div', {"class":"css-5wh65g"})

    # Ekstrak elemen web yang menampilkan nama produk, harga produk, nama toko, kota toko,jumlah terjual, dan rating dengan memanggil nama class-nya
    # try dan except untuk menghandling error jika data kosong sehingga bisa ditambahkan data None
    for produk in list_produk:
        try:
            nama = produk.find('div', {"class":"VKNwBTYQmj8+cxNrCQBD6g=="}).get_text()
            nama_produk.append(nama)
        except:
            nama_produk.append(None)
        
        try:
            harga = produk.find('div', {"class":"_8cR53N0JqdRc+mQCckhS0g=="}).get_text() 
            harga_produk.append(harga)
        except:
            harga_produk.append(None)
        
        try:
            penjual = produk.find('span', {"class":"X6c-fdwuofj6zGvLKVUaNQ=="}).get_text()  # Nama produk
            list_penjual.append(penjual)
        except:
            list_penjual.append(None)
        
        try:
            kota = produk.find('span', {"class":"-9tiTbQgmU1vCjykywQqvA== flip"}).get_text()  # Nama produk
            kota_toko.append(kota)
        except:
            kota_toko.append(None)
        
        try:
            terjual = produk.find('span', {"class":"eLOomHl6J3IWAcdRU8M08A=="}).get_text()  # Nama produk
            banyak_terjual.append(terjual)
        except:
            banyak_terjual.append(None)

        try:
            rating = produk.find('span', {"class":"nBBbPk9MrELbIUbobepKbQ=="}).get_text()  # Nama produk
            rating_produk.append(rating)
        except:
            rating_produk.append(None)

In [4]:
# Menutup driver
driver.quit()

In [2]:
import pandas as pd

In [5]:

# Membuat DataFrame dari semua kolom yang telah di scrapping
data_list = list(zip(nama_produk, harga_produk, list_penjual, kota_toko, banyak_terjual, rating_produk))
data = pd.DataFrame(data_list,columns=['nama_produk', 'harga_produk', 'penjual', 'kota_toko', 'banyak_terjual', 'rating_produk'])

### B. Data Preparation

In [6]:
data

nama_produk harga_produk  \
0    Geli Food Seblak Ceker Tanpa Tulang Super Peda...     Rp29.000   
1          Seblak instan komplit / seblak instan murah      Rp5.000   
2    KERUPUK SEBLAK RAFAEL PEDAS DAUN JERUK - 100 gram      Rp4.500   
3    KERUPUK SEBLAK MENTAH 1KG WARNA WARNI - Bawang...     Rp21.000   
4              Seblak Rafael, Seblak Coet Instan Halal     Rp25.000   
..                                                 ...          ...   
807       Kylafood Mix Pentol Kuah dan Seblak Original     Rp42.500   
808  Pasar Keripik Kerupuk Seblak Mawar Pedas size ...     Rp65.000   
809  Kylafood Mix Fav Selebgram Seblak Original & B...     Rp42.500   
810  Mie Seblak Geprek Sami Raos Khas Garut Seblak ...     Rp18.900   
811                       Kylafood Seblak Kwetiaw play     Rp19.975   

                   penjual        kota_toko banyak_terjual rating_produk  
0                 Lidigeli       Kab. Garut   100+ terjual           4.9  
1             ismileusa_id     Kab. Bandung   750+ terjual           4.8  
2               BociKakang  Jakarta Selatan    30+ terjual           4.8  
3               BociKakang  Jakarta Selatan   500+ terjual           5.0  
4        Brother Meat Shop  Jakarta Selatan   750+ terjual           5.0  
..                     ...              ...            ...           ...  
807               kylafood          Bandung    60+ terjual           4.9  
808  Pasar Keripik Bandung          Bandung     11 terjual           4.9  
809               kylafood          Bandung   250+ terjual           4.8  
810             Foodstocks  Jakarta Selatan   100+ terjual           5.0  
811               kylafood          Bandung   500+ terjual           4.9  

[812 rows x 6 columns]

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 812 entries, 0 to 811
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   nama_produk     812 non-null    object
 1   harga_produk    812 non-null    object
 2   penjual         812 non-null    object
 3   kota_toko       812 non-null    object
 4   banyak_terjual  777 non-null    object
 5   rating_produk   736 non-null    object
dtypes: object(6)
memory usage: 38.2+ KB


Terlihat data mentah dari web scrapping masih kotor:
- Karena ada beberapa toko baru, maka diambil beberapa data yang tidak memiliki banyak terjual dan rating produk. Baris data untuk toko - toko ini perlu dibuang.
- Kolom harga produk, banyak terjual, dan rating produk masih memiliki tipe yang salah, yaitu String. Hal ini karena angka data masih terdapat aksesoris seperti tanda `Rp`, `.`, `+`, `rb`, dan `terjual`. Harga produk dan banyak terjual perlu diganti menjadi `Integer`, sedangkan rating produk menjadi `Float`.
- Karena nama produk dan nama toko merupakan inputan dari user, maka ada potensi tidak rapi, perlu dilakukan penghilangan whitespace

In [3]:
import numpy as np

In [8]:
# Menghapus baris yang memiliki nilai None dan indeks diperbaiki
data = data.replace(to_replace='None', value=np.nan).dropna()
data.reset_index(drop=True, inplace=True)

In [9]:
# Menghapus whitespace pada nama produk dan penjual
data['nama_produk'] = data['nama_produk'].str.strip()
data['penjual'] = data['penjual'].str.strip()

In [10]:
# Menghapus Rp dan . pada kolom harga_produk.
data['harga_produk'] = data['harga_produk'].str.replace('Rp', '')
data['harga_produk'] = data['harga_produk'].str.replace('.', '')

# mengganti tipe kolom harga produk menjadi integer
data['harga_produk'] = data['harga_produk'].astype(int)

In [11]:
# Menghapus tanda +, terjual, dan mengganti rb menjadi 000
data['banyak_terjual'] = data['banyak_terjual'].str.replace('+', '')
data['banyak_terjual'] = data['banyak_terjual'].str.replace(' terjual', '')
data['banyak_terjual'] = data['banyak_terjual'].str.replace('rb', '000')

# Mengganti tipe kolom banyak terjual menjadi integer
data['banyak_terjual'] = data['banyak_terjual'].astype(int)

In [12]:
# Mengganti tipe kolom rating produk menjadi float
data['rating_produk'] = data['rating_produk'].astype(float)

In [13]:
data

nama_produk  harga_produk  \
0    Geli Food Seblak Ceker Tanpa Tulang Super Peda...         29000   
1          Seblak instan komplit / seblak instan murah          5000   
2    KERUPUK SEBLAK RAFAEL PEDAS DAUN JERUK - 100 gram          4500   
3    KERUPUK SEBLAK MENTAH 1KG WARNA WARNI - Bawang...         21000   
4              Seblak Rafael, Seblak Coet Instan Halal         25000   
..                                                 ...           ...   
731       Kylafood Mix Pentol Kuah dan Seblak Original         42500   
732  Pasar Keripik Kerupuk Seblak Mawar Pedas size ...         65000   
733  Kylafood Mix Fav Selebgram Seblak Original & B...         42500   
734  Mie Seblak Geprek Sami Raos Khas Garut Seblak ...         18900   
735                       Kylafood Seblak Kwetiaw play         19975   

                   penjual        kota_toko  banyak_terjual  rating_produk  
0                 Lidigeli       Kab. Garut             100            4.9  
1             ismileusa_id     Kab. Bandung             750            4.8  
2               BociKakang  Jakarta Selatan              30            4.8  
3               BociKakang  Jakarta Selatan             500            5.0  
4        Brother Meat Shop  Jakarta Selatan             750            5.0  
..                     ...              ...             ...            ...  
731               kylafood          Bandung              60            4.9  
732  Pasar Keripik Bandung          Bandung              11            4.9  
733               kylafood          Bandung             250            4.8  
734             Foodstocks  Jakarta Selatan             100            5.0  
735               kylafood          Bandung             500            4.9  

[736 rows x 6 columns]

In [14]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 736 entries, 0 to 735
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   nama_produk     736 non-null    object 
 1   harga_produk    736 non-null    int32  
 2   penjual         736 non-null    object 
 3   kota_toko       736 non-null    object 
 4   banyak_terjual  736 non-null    int32  
 5   rating_produk   736 non-null    float64
dtypes: float64(1), int32(2), object(3)
memory usage: 28.9+ KB


- Baris dengan nilai None dihapus sehingga data berjumlah 736
- Tipe data sudah benar untuk harga produk, banyak terjual, dan rating produk

In [15]:
# Export data menjadi csv
data.to_csv('data_gc3.csv', index=False)

In [5]:
# Membaca csv yang telah diexport untuk mendapatkan DataFrame data
data = pd.read_csv('catherine_kezia.csv')

### C. Business Understanding/Problem Statement

Untuk mengambil keputusan bisnis yang tepat, data perlu diberikan wawasan yang bermanfaat. Wawasan ini bisa didapat dengan mengikuti prinsip business development SMART (Specific, Measurable, Achievable, Relevant, Time-bound):
- `Specific`: Mengidentifikasi trend penjualan produk seblak di Tokopedia, termasuk jumlah penjualan, harga produk, daerah penjualan, dan rating-nya.

- `Measurable`: Mengumpulkan setidaknya 700 data produk seblak di Tokopedia, kemudian menganalisisnya untuk menemukan apakah produk telah dijual dengan jumlah yang banyak dan mendapatkan rating yang relatif positif. Produk dengan rating yang tinggi seperti 5.0 atau 4.9 dapat diukur untuk menentukan kualitas produk yang paling disukai customer.

- `Achievable`: melakukan web scraping dengan tools pada Python yang sesuai seperti BeautifulSoup atau Selenium. Data yang didapat dianalisis menggunakan Pandas dan Scipy

- `Relevant`: Fokus utama pada produk seblak karena sedangi trend di masyarakat. Hasil analisis akan berguna untuk memutuskan apakah seblak adalah produk yang sesuai untuk dropshipping.

- `Time-bound`: Proses pengumpulan dan analisis data dilakukan dalam jangka waktu 1-2 minggu, dengan hasil berupa laporan mengenai trend penjualan seblak di Tokopedia dan rekomendasi bisnis.


### D. Analysis

Sebelum melakukan analisis perlu dijabarkan pertanyaan - pertanyaan relevan mengenai penjualan seblak:
- `What`: Apakah seblak merupakan produk yang tepat dalam skema bisnis dropship?

- `Why`: Mengapa analisis ini penting dilakukan sebelum memulai bisnis dropship?

- `Who`: Siapa yang bisa menjadi target pasar produk ini?

- `Where`: Dimana tempat ideal produk seblak dijual, dengan mempertimbangkan harga produksi?

- `When`: Kapan data ini akan diambil dan dianalisis?

- `How`: Bagaimana persepsi customer terhadap produk seblak di Tokopedia?

#### 1. Analisis kolom harga produk, banyak terjual, dan rating produk

- Ingin dianalasis persebaran dan distribusi data untuk kolom harga produk, banyak terjual, dan rating produk

1.1 Analisis kolom harga produk

In [8]:
harga_avg = data['harga_produk'].mean()
print('Rata-rata harga: ', harga_avg)

harga_med = data['harga_produk'].median()
print('Median harga: ', harga_med)

harga_std = data['harga_produk'].std()
print('Standar deviasi harga: ', harga_std)

harga_skew = data['harga_produk'].skew()
print('Skewness harga: ', harga_skew)

harga_kurt = data['harga_produk'].kurtosis()
print('Kurtosis harga: ', harga_kurt)

Rata-rata harga:  20551.088315217392
Median harga:  14949.5
Standar deviasi harga:  22548.20378327402
Skewness harga:  4.57534673002601
Kurtosis harga:  37.20080763212851


- Rata - rata harganya lebih besar dari median. Ada beberapa harga dengan nilai ekstrim yang menarik rata - rata harga ke atas
- Standar deviasi `22548.20` termasuk sangat tinggi, terdapat fluktuasi harga yang signifikan pada dataset
- Skewness bernilai `4.58`, termasuk highly skewed. Distribusi harga sangat miring ke kanan, berarti sebagian besar data harga memiliki nilai yang rendah
- Kurtosis bernilai `37.20`, distribusi memiliki puncak yang tajam dan berekor panjang. Menunjukkan harga barang memiliki banyak nilai ekstrim dalam dataset

1.2 Banyak terjual

In [9]:
jumlah_avg = data['banyak_terjual'].mean()
print('Rata-rata terjual: ', jumlah_avg)

jumlah_med = data['banyak_terjual'].median()
print('Median terjual: ', jumlah_med)

jumlah_std = data['banyak_terjual'].std()
print('Standar deviasi terjual: ', jumlah_std)

jumlah_skew = data['banyak_terjual'].skew()
print('Skewness terjual: ', jumlah_skew)

jumlah_kurt = data['banyak_terjual'].kurtosis()
print('Kurtosis terjual: ', jumlah_kurt)

Rata-rata terjual:  440.6195652173913
Median terjual:  85.0
Standar deviasi terjual:  1217.753625807662
Skewness terjual:  5.249530284371333
Kurtosis terjual:  32.020500684866064


- Rata - rata jumlah terjual jauh lebih besar dari median. Ada beberapa nilai ekstrim yang menarik rata - rata jumlah terjual ke atas
- Standar deviasi `1217.75` termasuk sangat tinggi, menunjukkan variasi yang cukup ekstrim dalam jumlah terjualnya barang
- Skewness bernilai `5.24`, termasuk highly skewed. Distribusi jumlah terjual sangat miring ke kanan, berarti sebagian besar penjualan memiliki jumlah yang rendah
- Kurtosis bernilai `32.02`, distribusi memiliki puncak yang tajam dan berekor panjang. Menunjukkan jumlah produk terjual memiliki banyak nilai ekstrim dalam dataset, sehingga distribusi data disimpulkan tidak normal

1.3 Rating Produk

In [10]:
rating_avg = data['rating_produk'].mean()
print('Rata-rata rating: ', rating_avg)

rating_med = data['rating_produk'].median()
print('Median rating: ', rating_med)

rating_std = data['rating_produk'].std()
print('Standar deviasi rating: ', rating_std)

rating_skew = data['rating_produk'].skew()
print('Skewness rating: ', rating_skew)

rating_kurt = data['rating_produk'].kurtosis()
print('Kurtosis rating: ', rating_kurt)

Rata-rata rating:  4.874592391304348
Median rating:  4.9
Standar deviasi rating:  0.1675549568382758
Skewness rating:  -3.0991384541933744
Kurtosis rating:  14.54690216284946


- Mayoritas produk mendapatkan rating yang sangat tinggi, mendekati 5
- Median mendekati rata-rata, menandakan distribusi yang cukup konsisten pada rating yang tinggi, yaitu sekitar 4.9
- Standar deviasi sebesar `0.17`, menunjukkan variabilitas rating sangat rendah. Sebagian besar rating berada sangat dekat dengan rata-rata
- Skewness bernilai `-3.09`, termasuk highly skewed. Distribusi rating sangat miring ke kiri, berarti sebagian besar penjualan memiliki nilai rating yang tinggi (mendekati 5)
- Kurtosis bernilai `32.02`, distribusi memiliki puncak yang tajam dan berekor panjang. Rating berkumpul di sekitar nilai rata-rata (di puncak), tetapi ada beberapa nilai ekstrim yang lebih rendah.

#### 2. Berapa maksimum dan minimum profit yang bisa didapat dalam penjualan seblak di Tokopedia?

Meskipun didapat datanya skewed akan dianggap datanya berdistribusi normal dengan confidence level 95%

In [6]:
from scipy import stats

In [11]:
avg_pendapatan = jumlah_avg * harga_avg
std_pendapatan = jumlah_std * harga_std

N = len(data)

low, up = stats.norm.interval( 
    0.95,                 
    loc=avg_pendapatan,
    scale=std_pendapatan/np.sqrt(N)
    )
print('Lower Limit:',low)   # Lower limit
print('Upper Limit:',up)
avg_pendapatan

Lower Limit: 7071490.710081948
Upper Limit: 11038932.486308647


9055211.598195298

Dengan confidence 95%, pendapatan bulanan dari penjualan seblak akan berada di antara Rp7,071,490 dan Rp11,038,932

#### 3.  Apakah daerah penjualan baik itu di dalam atau di luar Jabodetabek bisa mempengaruhi harga produk?

Hipotesis yang perlu dianalisi adalah:



H0: harga barang di Jabodetabek = harga barang di luar Jabodetabek

H1: harga barang di Jabodetabek != harga barang di luar Jabodetabek

In [13]:
data['kota_toko'].unique()

array(['Kab. Garut', 'Kab. Bandung', 'Jakarta Selatan', 'Surabaya',
       'Bandung', 'Cimahi', 'Kab. Bogor', 'Tasikmalaya', 'Jakarta Barat',
       'Jakarta Timur', 'Jakarta Pusat', 'Tangerang Selatan',
       'Kab. Klaten', 'Kab. Tangerang', 'Kab. Indramayu', 'Kediri',
       'Banjarbaru', 'Kab. Bekasi', 'Kab. Magelang', 'Jakarta Utara',
       'Surakarta', 'Kab. Sidoarjo', 'Kab. Majalengka', 'Depok',
       'Kab. Tasikmalaya', 'Kab. Bantul', 'Kab. Banyuwangi', 'Tangerang',
       'Kab. Bandung Barat', 'Medan', 'Kab. Tulungagung', 'Kab.Ciamis',
       'Kab. Karawang', 'Kab. Badung', 'Kab. Cirebon', 'Kab. Sukabumi',
       'Kab. Jember', 'Bekasi', 'Kab. Kebumen', 'Pekanbaru', 'Malang',
       'Kab. Cianjur', 'Kab. Mojokerto', 'Semarang', 'Kab. Kediri',
       'Palembang', 'Bandar Lampung', 'Makassar', 'Bogor', 'Banjarmasin',
       'Kab. Sumedang', 'Kab. Boyolali', 'Kab. Purworejo', 'Sukabumi',
       'Kab. Jombang', 'Kab. Sukoharjo', 'Kab. Purwakarta',
       'Kab. Banyumas', 'Kab. M

In [15]:
# Menyimpan semua nama ota dalam jabodetabek
jabodetabek = ['Jakarta Selatan', 'Jakarta Barat', 'Jakarta Timur', 'Jakarta Pusat', 'Tangerang Selatan', 'Kab. Tangerang',
               'Kab. Bekasi', 'Jakarta Utara', 'Tangerang', 'Depok', 'Bekasi', 'Kab. Cianjur', 'Bogor',
               ]

# Menyimpan semua data seblak di dalam dan di luar jabodetabek
data_jabodetabek = data[data['kota_toko'].isin(jabodetabek)]
data_luar_jabodetabek = data[~data['kota_toko'].isin(jabodetabek)]

In [16]:
data_jabodetabek

nama_produk  harga_produk  \
2    KERUPUK SEBLAK RAFAEL PEDAS DAUN JERUK - 100 gram          4500   
3    KERUPUK SEBLAK MENTAH 1KG WARNA WARNI - Bawang...         21000   
4              Seblak Rafael, Seblak Coet Instan Halal         25000   
10   KERUPUK MENTAH KERUPUK SEBLAK MENTAH 500 GR - ...         13500   
12              KERUPUK SEBLAK PEDAS DAUN JERUK 100 GR          4500   
..                                                 ...           ...   
723  Seblak Kencur 250gr/Kerupuk Seblak Rasa Kencur...         12500   
724          KERUPUK SEBLAK KERING 1KG BY OYUN KRIUKZZ         47400   
726                                     kerupuk seblak          8000   
730       Makanan Kering Kerupuk Seblak Pedas 100 gram          5000   
734  Mie Seblak Geprek Sami Raos Khas Garut Seblak ...         18900   

               penjual        kota_toko  banyak_terjual  rating_produk  
2           BociKakang  Jakarta Selatan              30            4.8  
3           BociKakang  Jakarta Selatan             500            5.0  
4    Brother Meat Shop  Jakarta Selatan             750            5.0  
10          BociKakang  Jakarta Selatan             250            4.9  
12          BociKakang  Jakarta Selatan             750            4.8  
..                 ...              ...             ...            ...  
723   Agen Snack Murah    Jakarta Barat             250            5.0  
724       oyun kriukzz    Jakarta Barat              30            5.0  
726         food house    Jakarta Timur              30            5.0  
730             BLIBET    Jakarta Barat              11            5.0  
734         Foodstocks  Jakarta Selatan             100            5.0  

[240 rows x 6 columns]

In [17]:
data_luar_jabodetabek

nama_produk  harga_produk  \
0    Geli Food Seblak Ceker Tanpa Tulang Super Peda...         29000   
1          Seblak instan komplit / seblak instan murah          5000   
5                   Kerupuk Mentah 250gr Seblak Mentah          5900   
6                     Kylafood Seblak Rempah Authentik         10965   
7                  seblak krupuk 5000 ,pedas dan gurih          5000   
..                                                 ...           ...   
729  Kerupuk Seblak Oranye (200gr) - Kerupuk Kempla...          4000   
731       Kylafood Mix Pentol Kuah dan Seblak Original         42500   
732  Pasar Keripik Kerupuk Seblak Mawar Pedas size ...         65000   
733  Kylafood Mix Fav Selebgram Seblak Original & B...         42500   
735                       Kylafood Seblak Kwetiaw play         19975   

                   penjual     kota_toko  banyak_terjual  rating_produk  
0                 Lidigeli    Kab. Garut             100            4.9  
1             ismileusa_id  Kab. Bandung             750            4.8  
5              Pedas Gilaa      Surabaya            1000            5.0  
6                 kylafood       Bandung            3000            4.9  
7           anugrah seblak  Kab. Bandung            1000            4.8  
..                     ...           ...             ...            ...  
729    Pusat Grosir Cuanki  Kab. Bandung              11            5.0  
731               kylafood       Bandung              60            4.9  
732  Pasar Keripik Bandung       Bandung              11            4.9  
733               kylafood       Bandung             250            4.8  
735               kylafood       Bandung             500            4.9  

[496 rows x 6 columns]

Jenis analisa sama atau tidak sama yang tidak ada ketergantungan cocok menggunakan Two-Sample Independent Test

In [20]:
t_stat, p_value = stats.ttest_ind(data_jabodetabek['harga_produk'], data_luar_jabodetabek['harga_produk'])

print(f"t-statistic: {t_stat}")
print(f"p-value: {p_value}")

t-statistic: -1.5383693201308555
p-value: 0.12438929482345605


- Didapat nilai `t-statistic` -1.538, menunjukkan bahwa perbedaan harga antara seblak di jabodetabek dan di luar jabodetabek relatif kecil
- Didapat nilai `p-value` 0.124, berarti lebih besar dari tingkat signifikan yang bernilai 0.05. Maka H0 diterima dan disimpulkan bahwa harga barang di Jabodetabek dan luar Jabodetabek sama.


#### 4.  Apakah jumlah harga bisa mempengaruhi rating yang didapat untuk produk?

Sama dengan proses menentukan upper bound dan lower bound, data akan dianggap sebagai data berdistribusi normal, sehingga menggunakan uji kolerasi Pearson

Hipotesis yang perlu dianalisa adalah:

H0: Tidak ada hubungan antara harga produk dengan rating produk

H1: Ada hubungan antara harga produk dengan jumlah rating produk

In [25]:
# Uji korelasi Pearson
corr_pearson, p_value_pearson = stats.pearsonr(data['harga_produk'], data['rating_produk'])
print(f"Korelasi Pearson: {corr_pearson}")
print(f"P-value Pearson: {p_value_pearson}")

Korelasi Pearson: 0.007372366485596757
P-value Pearson: 0.8417386383461116


- Korelasi Pearson bernilai `0.007`. Karena mendekati 0, menunjukkan bahwa harga produk dan rating produk tidak memiliki hubungan linear.  Customer tidak cenderung memberikan rating yang lebih tinggi atau lebih rendah berdasarkan harga produk.
- Didapat p-value `0.84`jauh lebih besar daripada tingkat signifikansi yaitu `0.05`. Hipotesis 0 gagal ditolak, tidak ada hubungan yang signifikan antara harga dan rating produk

### E. Kesimpulan

Dari analisis data yang telah dilakukan, maka bisa menjawab pertanyaan - pertanyaan yang telah dijabarkan:

- Seblak bisa menjadi produk yang baik untuk skema bisnis dropship. Bisa dilihat dari fleksibilitas harga barang dan lokasi penjualan. Hasil analisis menunjukkan bahwa harga barang tidak mempengaruhi rating kepuasaan customer. Dan juga tidak perlu lokasi penjualan yang spesifik karena harga barang di area baik itu dalam maupun luar Jabodetabek relatif sama, berarti bisa disimpulkan juga bahwa harga produksi juga relatif sama.

- Analisa ini bisa membantu keputusan bisnis dengan memberikan informasi tentang minat customer pada produk, dan juga faktor apa saja yang bisa mempengaruhi minat tersebut. Bisa dilihat dari rating produk seblak di Tokopedia berada di sekitar `4.9`, nilai ini sangat tinggi maka bisa disimpulkan masyarakat menyukai jenis produk ini. Juga bisa membantu dalam menunjukkan perkiraan profit yang bisa didapat, hasil analisis menyimpulkan bahwa bisa mendapat sekitar `Rp7,071,490` sampai `Rp11,038,932`.

- Dilihat dari hasil analisa hubungan rating dengan harga yang menunjukkan tidak ada hubungan linear, dan juga harga produk dengan lokasi yang menunjukkan tidak ada perbedaan signifikan, maka produk ini bisa untuk customer dengan kondisi ekonomi apapun dan juga di lokasi mana saja.

- Dengan mempertimbangkan harga produksi, didapat bahwa harga barang tidak jauh berbeda, seblak bisa dijual di dalam maupun di luar wilayah jabodetabek

- Hasil analisa berikut dianggap sebagai informasi penjualan produk per-bulan, dan bisa mendapat sebanyak 700 data yang layak untuk dianalisis. Maka cocok untuk mengambil dan menganalisa data per-bulan untuk mendapat data yang cukup

- Dilihat dengan rating yang berkisar tinggi, profit yang bisa didapat dari jumlah produk yang terjual cukup tinggi maka produk seblak bisa dianggap memiliki persepsi yang baik diantara customer Tokopedia.

Keputusan akhir dari analisa adalah:

Produk seblak layak untuk dijual pada skema bisnis dropship karena memiliki persepsi yang baik dari antara customer dan juga memiliki fleksibilitas yang tinggi dalam melakukan penjualan